lib/rules/sotd/charter-disclosure.js

```diff
@@ -1,15 +1,81 @@
+var sua = require("../../throttled-ua");
+
 const self = {
     name: 'sotd.charter-disclosure'
 };
 
 exports.name = self.name;
 
+var charterText = /The disclosure obligations of the Participants of this group are described in the charter\./;
+
 exports.check = function (sr, done) {
     var $sotd = sr.getSotDSection();
     if ($sotd && $sotd.length) {
+
+        // check text exists
         var txt = sr.norm($sotd.text());
-        if (!/The disclosure obligations of the Participants of this group are described in the charter\./.test(txt))
-            sr.error(self, "not-found");
+        if (!charterText.test(txt)) {
+            sr.error(self, "text-not-found");
+            return done();
+        }
+
+        // check "charter" link is found and correct
+        var charterLinkFound = false;
+        $sotd.find("a[href]").each(function () {
+            var $a = sr.$(this)
+                , charterHref = $a.attr("href")
+                , text = sr.norm($a.text())
+                , pText = sr.norm($a.parent().text())
+                ;
+
+            // Find the right paragraph and right link.
+            if (charterText.test(pText) && text == "charter") {
+                charterLinkFound = true;
+                
+                var expectedHref
+                    , deliverId = sr.getDelivererIDs()
+                    , url = 'https://api.w3.org/groups/' + deliverId[0] + '/charters'
+                    , req
+                    , apikey = process.env.W3C_API_KEY
+                    , ua = "W3C-Pubrules/" + sr.version
+                    ;
+                if (deliverId.length === 0) {
+                    sr.error(self, "no-group");
+                    return done();
+                }
```

Comment:
This should be checked before the `.each`. No need to go through all the SotD if there's no deliverer id and this will simplify the handling of `done()` because here, you aren't finishing the rule, you just break the current each loop.